styx-common/src/main/java/com/spotify/styx/storage/DatastoreStorage.java

```diff
@@ -154,20 +162,24 @@
   public static final boolean DEFAULT_CONFIG_RESOURCES_SYNC_ENABLED = false;
 
   public static final int MAX_RETRIES = 100;
+  public static final int MAX_NUMBER_OF_ENTITIES_IN_ONE_BATCH = 1000;
 
   private final Datastore datastore;
   private final Duration retryBaseDelay;
   private final Function<Transaction, DatastoreStorageTransaction> storageTransactionFactory;
+  private final ForkJoinPool forkJoinPool;
 
   DatastoreStorage(Datastore datastore, Duration retryBaseDelay) {
     this(datastore, retryBaseDelay, DatastoreStorageTransaction::new);
   }
 
+  @VisibleForTesting
   DatastoreStorage(Datastore datastore, Duration retryBaseDelay,
-      Function<Transaction, DatastoreStorageTransaction> storageTransactionFactory) {
+                   Function<Transaction, DatastoreStorageTransaction> storageTransactionFactory) {
     this.datastore = Objects.requireNonNull(datastore);
     this.retryBaseDelay = Objects.requireNonNull(retryBaseDelay);
     this.storageTransactionFactory = Objects.requireNonNull(storageTransactionFactory);
+    this.forkJoinPool = new ForkJoinPool();
```

Comment:
Didn't bother injecting which doesn't bring much value in this case. FJP is quite deterministic.

Reply:
It seems undesirable to have the datastore IO parallelism be derived from the number of cpu cores on the host.

Specify an explicit level of desired parallelism.